### PR TITLE
[Dashboard] Feat: Add members and followers to colony header

### LINKED ISSUE #2847: [Dashboard] Update Dashboard colony header to include members and followers
### Story

As a USER, I would like to see an overview of members and followers as a part of the colony.

### Description

Given the focus on financial actions and tools, the colony dashboard header should be adjusted to include the members and followers. This implies the colony description is removed. 
If a new colony is created, there will be one member by default.

We must display maximum 4 random members avatars.

The difference between a member and a follower is: a member is someone with reputation or permissions, while a follower is someone who joined the colony.

### This issue is complete when...

- [ ] Dashboard header includes maximum 4 avatars and total number of members
- [ ] Dashboard header includes total number of followers
- [ ] Dashboard header doesn't show up the colony description

[Figma link](https://www.figma.com/design/gac6xVgGNCAge5Bia933dp/Dashboard?node-id=4859-36367&t=3YmH0OCdhMXflaYC-0)
[Figma link empty state](https://www.figma.com/design/gac6xVgGNCAge5Bia933dp/Dashboard?node-id=4867-40639&t=3YmH0OCdhMXflaYC-0)

**Feature branch**: `feat/enhanced-dashboard`

### PR DESCRIPTION
**Pull request by iamsamgibbs**
## Description

As part of the dashboard redesign, the members widget is being removed from the dashboard and moved into the dashboard header. There is now a followers count in addition to the members count. A member is someone with reputation or permissions (`hasReputation: true || hasPermissions: true`), while a follower is someone who joined the colony (`isWatching: true`).

## Testing

* Step 1 - Load up the colony dashboard. After running the create-data script you should expect to see 4 members and 32 followers.

Screenshot 2024-08-29 at 14 28 02

* Step 2 - Check the components (including the loading skeleton) match the [figma design
](https://www.figma.com/design/gac6xVgGNCAge5Bia933dp/Dashboard?node-id=4859-36367&t=3YmH0OCdhMXflaYC-0)
* Step 3 - Navigate to [http://localhost:9091/planex/members/contributors](http://localhost:9091/planex/members/contributors) to check which 4 members have reputation.
* Step 4 - Assign permissions in Andromeda to any member without reputation.

Screenshot 2024-08-29 at 14 30 58

* Step 5 - Refresh the dashboard. The members should increase to 5.
* Step 6 - Award reputation to any members without reputation (and not the same member as step 3).

Screenshot 2024-08-29 at 14 32 25

* Step 7 - Turn on the reputation monitor: [http://127.0.0.1:3001/reputation/monitor/toggle](http://127.0.0.1:3001/reputation/monitor/toggle)
* Step 8- Forward time with `npm run forward-time 24`
* Step 9 - Refresh the dashboard. The members should increase to 6.
* Step 10 - Select 'Andromeda' from the team switcher. The members should be 5.
* Step 11 - Select any other sub team from the team switcher. The members should be 4.

Further testing:

* Step 1 - Create a new colony with `node scripts/create-data.js`
* Step 2 - Once the colony has been created, refresh the dashboard. The members and followers should both be 1. [Figma design](https://www.figma.com/design/gac6xVgGNCAge5Bia933dp/Dashboard?node-id=4867-40639&t=3YmH0OCdhMXflaYC-0)

Screenshot 2024-08-29 at 14 36 20

* Step 3 - Invite Amy to the colony:

```
query MyQuery {
  listColonies {
    items {
      name
      colonyMemberInviteCode
    }
  }
}
```

`http://localhost:9091/invite/COLONYNAME/INVITECODE`

* Step 4 - The number of followers should increase to 2 (might require a refresh after joining).

Screenshot 2024-08-29 at 14 37 08

* Step 5 - Switch back to Leela and assign permissions to Amy.
* Step 6 - Refresh the dashboard and the number of members should increase to 2.

Screenshot 2024-08-29 at 14 37 57

* Step 7 - Switch back to Amy and leave the colony.
* Step 8 - Switch back to Leela, and refresh the dashboard, the number of followers should decrease to 1.

Screenshot 2024-08-30 at 08 57 59

## Diffs

**New stuff** ✨

* Members information component added to colony header
* Added followers to MemberContext

**Changes** 🏗

* Add loading skeleton to UserAvatars

**Deletions** ⚰️

* Members dashboard widget removed
* Colony description removed from colony header

Resolves #2847

### CODE REVIEW

**Review by Nortsova — CHANGES_REQUESTED** [review]
Great work @iamsamgibbs 👏

I can confirm that after load create-data script I can see 4 members and 32 followers:
image

Members was added after Step 4 and Step 9:
image

For new colony I can see 1 member:
image

After I invited amy number of followers became 2:
image

After assigned amy permissions she became a member:
image

Skeleton looks great:
image

Here is a couple of comments that I observed, but please feel free to ignore if it is not related to this PR:

It seems that avatars in the design have more distance: 
image

And padding top should be a little bit more: 
image

**Comment by iamsamgibbs** [comment]
Thanks for the review @Nortsova ! Good catch on the avatar overlap distance, I'll fix that up.

I also noticed the top padding but felt it was out of scope as it effects a bunch of the pages not just the dashboard header.

**Review by Nortsova — APPROVED** [review]
image

Thank you for fixing this small issue, now it works perfectly! 👏

File: src/components/v5/frame/ColonyHome/partials/DashboardHeader/partials/MembersInformation/MembersInformation.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    filteredMembers,
+    followersCount,
+  } = useMemberContext();
+
+  const selectedMembers = nativeDomainId
+    ? filteredMembers.filter(
+        ({ roles, reputation }) =>
+          roles?.items?.find(
+            (role) => role?.domain.nativeId === nativeDomainId,
+          ) ||
```
Comment (mmioana) [inline src/components/v5/frame/ColonyHome/partials/DashboardHeader/partials/MembersInformation/MembersInformation.tsx:30]: Could we extract this `(role) => role?.domain.nativeId === nativeDomainId` into a more generic function

```
type EntityWithDomainNativeId = {
  domain: Pick<Domain, 'nativeId'>
};

const filterByDomainNativeId = (entity: EntityWithDomainNativeId | null, nativeDomainId) => entity?.domain.nativeId === nativeDomainId
```

And then use it `(role) => filterByDomainNativeId(role, nativeDomainId)`

File: src/components/v5/frame/ColonyHome/partials/DashboardHeader/partials/MembersInformation/MembersInformation.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          reputation?.items?.find(
+            (rep) => rep?.domain.nativeId === nativeDomainId,
+          ),
```
Comment (mmioana) [inline src/components/v5/frame/ColonyHome/partials/DashboardHeader/partials/MembersInformation/MembersInformation.tsx:33]: And use the generic function from above also here `(rep) => filterByDomainNativeId(rep, nativeDomainId),`

**Review by mmioana — APPROVED** [review]
Thank you for picking up this issue @iamsamgibbs 🙏 Amazing work as usual 🌟 

Can confirm all the cases in the testing steps work flawlessly ✨ 

After running `create-data` script
Screenshot 2024-09-02 at 08 36 29
Screenshot 2024-09-02 at 08 38 39

Managing permissions for a member without
Screenshot 2024-09-02 at 08 40 28
Number of members gets increased to `5`
Screenshot 2024-09-02 at 08 40 42

Awarding reputation
Screenshot 2024-09-02 at 08 41 24

Screenshot 2024-09-02 at 08 42 01

Number of members gets increased to `6`
Screenshot 2024-09-02 at 08 42 21


Switch to `Andromeda`
Screenshot 2024-09-02 at 08 42 37

Switch to `Rocinate`

Screenshot 2024-09-02 at 08 42 47


Creating a new colony
Screenshot 2024-09-02 at 08 43 56
Screenshot 2024-09-02 at 08 44 17
`amy` joins new colony
Screenshot 2024-09-02 at 08 45 16
Screenshot 2024-09-02 at 08 45 46
Adding permissions to `amy`
Screenshot 2024-09-02 at 08 46 53
Screenshot 2024-09-02 at 08 47 06
`amy` leaves the newly created colony 
Screenshot 2024-09-02 at 08 47 52

Left a couple of refactoring comments, but feel free to tackle them if you consider they make sense

**Review by rdig — APPROVED** [review]
Went through the whole testing flow and it behaves as expected, nicely done!

Was on my laptop and it's slower, so the reputation based contributors would load slower (ie: you'd see 5 then immediately refresh to 6)

But all's good, so this is ready to merge!

Screenshot from 2024-09-03 18-54-13
Screenshot from 2024-09-03 18-54-36
Screenshot from 2024-09-03 18-56-19
Screenshot from 2024-09-03 18-56-48
Screenshot from 2024-09-03 18-57-34
Screenshot from 2024-09-03 18-58-44
Screenshot from 2024-09-03 18-59-46

**Pull request merged**